tests/test_ccs.py

```diff
@@ -0,0 +1,53 @@
+import unittest
+try:
+    from unittest.mock import patch, ANY
+except:
+    from mock import patch, ANY
+
+from tests.http_client import MinimalResponse
+from tests.test_token_cache import build_response
+
+import msal
+
+
+class TestCcsRoutingInfoTestCase(unittest.TestCase):
+
+    def test_acquire_token_by_auth_code_flow(self):
+        app = msal.ClientApplication("client_id")
+        state = "foo"
+        flow = app.initiate_auth_code_flow(
+            ["some", "scope"], login_hint="johndoe@contoso.com", state=state)
+        with patch.object(app.http_client, "post", return_value=MinimalResponse(
+                status_code=400, text='{"error": "mock"}')) as mocked_method:
+            app.acquire_token_by_auth_code_flow(flow, {"state": state, "code": "bar"})
```

Comment:
You can also request the client info from the /authorize endpoint by adding adding "client_info=1" and use that to create the "Oid:foo@bar" header that can be sent to the token endpoint so there should probably be a test for that too.

Reply:
Thanks for the hint! The `client_info=1` trick was not mentioned in the body of [internal design](https://identitydivision.visualstudio.com/DevEx/_git/AuthLibrariesApiReview/pullrequest/3304), but only in a conversation which I overlooked. Now I refactor this PR to request and utilize `client_info`. My test cases are also updated.

Reply:
FYI: I documented that usage into our internal specs here. https://identitydivision.visualstudio.com/DevEx/_git/AuthLibrariesApiReview/pullrequest/3964